Symfony 3.3 upgrade

danrot opened the pull request:
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | ---
| Related issues/PRs | ---
| License | MIT
| Documentation PR | ---

#### What's in this PR?

This PR fixes a few issues that came up with Symfony 3.3:

* `key` on a top level argument is not allowed anymore
* Server commands have been moved, and we inherited from them. Removed our own implementation.
* Set-Cookie headers are now also cached

#### Why?

Because we want to stay compatible.

danrot commented on src/Sulu/Bundle/AudienceTargetingBundle/EventListener/TargetGroupSubscriber.php:
```diff
@@ -180,7 +180,7 @@ public function setTargetGroup(GetResponseEvent $event)
                 $this->targetGroupStore->updateTargetGroupId($targetGroup->getId());
             }
-        } else {
+        } elseif ($request->getPathInfo() !== $this->targetGroupUrl) {
```
add comment to explain what's happening here.

weaverryan commented:
Hey guys! I wasn't sure if you knew, the server commands live in WebServerBundle now - so they weren't entirely removed - just moved to that bundle.

Cheers!

danrot commented:
@weaverryan Thank you for the hint, but I was aware of that, just wrote it a little bit confusing 🙈 The problem was that we inherited from these commands, and it would be hard to do that now and stay compatible with Symfony 3.3 and lower versions at the same time.

We originally introduced these commands because we needed two different kernels running at the same time, but we fixed this in a different manner now (see https://github.com/sulu/sulu-standard/pull/823/files#diff-64a0fd926147fcc158821dcb6f1b4d11), so we don't need our own implementations anymore. So I removed them, that's what I actually meant in the description 😄

weaverryan commented:
@danrot beautiful - thanks for the quick update (and apologies for the surprise issues!)

danrot commented:
@weaverryan Don't worry, we are also maintaining an open source project, have experienced how hard it is to keep BC 🙈
